share/normalize/links.py

```diff
@@ -368,7 +409,37 @@ def execute(self, obj):
 
 class TextLink(AbstractLink):
     def execute(self, obj):
-        return obj.text
+        if isinstance(obj, str):
+            return obj
+        elif isinstance(obj, dict):
+            if '#text' in obj:
+                return obj['#text']
+            raise Exception('#text is not in {}'.format(obj))
+        elif isinstance(obj, list):
+            for item in obj:
+                if isinstance(item, str):
+                    return item
+            raise logger.warning('No value in list {} is a string.'.format(obj))
```

Comment:
I think this is actually a better example:`tools.Try(tools.Text(ctx.nameIdentifier))`

`nameIdentifier` won't always be there but we don't wan't everything to break if it's not there.